blueocean-dashboard/src/main/js/redux/actions.js

```diff
@@ -57,7 +57,23 @@ export const actionHandlers = {
         return state.set('currentRuns', null);
     },
     [ACTION_TYPES.SET_CURRENT_RUN_DATA](state, { payload }): State {
-        return state.set('currentRuns', payload);
+        return state.set('currentRuns', payload.map((run) => {
+            if (run._class === 'io.jenkins.blueocean.service.embedded.rest.QueueItemImpl') {
```

Comment:
Sure, can use the capabilities API, but my point is that while the `_class` is visible in the response then people will take the "easy" option and just do a string compare on that and not bother figuring out how to use the capabilities API.